src/pfe/file-watcher/server/test/README.md

```diff
@@ -0,0 +1,104 @@
+
+To run the functional test on kube, run the following commands:
+- `cd src/pfe/file-watcher/server/test/`
+- `CLUSTER_IP=<cluster_ip> CLUSTER_PORT=<cluster_port> CLUSTER_USER=<cluster_user> CLUSTER_PASSWORD=<cluster_password> ./test.sh -t kube -c y -p y`
+
+The options `-t` takes in the type of test environment to run, by default it is `local`. The options `-c` and `-p` stands for clean up before running and post clean up after test has finished.
+
+Additional options can be passed such as `IMAGE_PUSH_REGISTRY_ADDRESS`, `IMAGE_PUSH_REGISTRY_NAMESPACE` and `USER_DEVFILE` to overwrite the default values. 
+
+The default values are:
+- **IMAGE_PUSH_REGISTRY_ADDRESS**: docker-registry.default.svc:5000/che
+- **IMAGE_PUSH_REGISTRY_NAMESPACE**: eclipse-che
+- **USER_DEVFILE**: https://raw.githubusercontent.com/eclipse/codewind-che-plugin/master/devfiles/latest/devfile.yaml
+
+Don't have a Che instance? No worries, use the `che-setup.sh` script from https://github.com/eclipse/codewind-che-plugin/tree/master/scripts to deploy a che instance with codewind. More on how to deploy che automatically on the link above.
+
+## Developing new test
+
+### Unit Test Suite
+
+To add new test cases under `unit-test`, go under `src/pfe/file-watcher/server/test/unit-test`. The main file that kicks of the test is `unit-test.ts` and subsequent test files are stored under `src/pfe/file-watcher/server/test/unit-test/tests`.
+
+### Functional Test Suite
+
+To add new test cases under `functional-test`, go under `src/pfe/file-watcher/server/test/functional-test`. The main file that kicks of the test is `functional-test.ts` and subsequent test suites are stored under `src/pfe/file-watcher/server/test/functional-test/suites`.
+
+There are two types of suites:
+- **Generic**: This suite has all the generic turbine functionalities such as setting locales, workspace settings and log level settings.
+- **Project**: This suite has all the project specific functionalities such as create, delete, update and etc.
+
+Generic suite is ran only once during the functional run whereas the project specific suite is ran across all supported project types.
+
+To run the generic suite only:
+- set `genericSuite.runTest(true);` in **runAllTests** functions in `src/pfe/file-watcher/server/test/functional-test/functional-test.ts`
+
+To run the project specific suite only:
+- set `projectSuite.runTest(projData, chosenTemplate, chosenProject, true);` in **runAllTests** functions in `src/pfe/file-watcher/server/test/functional-test/functional-test.ts`
+
+Specific project types supported:
+- **Codewind Default**
+  - Liberty
+  - Node.js
+  - Spring
+  - Swift
+- **Generic Docker**
+  - Go
+  - Python
+  - Lagom
+- **OpenShift DO** (kube only)
+  - Node.js
+  - Perl
+  - Python
+
+To run only specific project types, comment out the project types not unwanted in **projectTypes** variable under `src/pfe/file-watcher/server/test/functional-test/configs/app.config.ts`.
```

Comment:
`project types not unwanted` should be `unwanted project types`?

Reply:
Done